more/auction/agencyEscrow.js

```diff
@@ -0,0 +1,119 @@
+/* global E makePromise */
+// Copyright (C) 2019 Agoric, under Apache License 2.0
+
+import harden from '@agoric/harden';
+import { mustBeSameStructure } from '../../util/sameStructure';
+
+// There are two parties to this transaction. The buyer is offering some amount
+// of currency (any fungible good) for a valuable item. The buyer will either
+// get the item, and possibly a return of some currency, or will get all their
+// offer back. The agency needs to see the offer before deciding whether to
+// transact. Only the amount is visible to the agency before an escrow seat is
+// received for the transaction.
+const agencyEscrow = {
+  start: (terms, inviteMaker) => {
+    const { currencyAmount, goodsAmount } = terms;
+    const { issuer: goodsIssuer } = goodsAmount.label;
+
+    // We want to give the buyer a promise for the good and for a refund. The
+    // refund will resolve either to all the buyer's deposit, or the portion of
+    // the deposit that wasn't required. The agency will get an invite for a
+    // seat that can ask the amount deposited, and then either cancel or provide
+    // an escrow seat with a specific price up to the amount.
+
+    // Buyer's winnings and refund will be resolved if the offer is consummated.
+    // If it's cancelled, only refund will be resolved. Winnings will contain
+    // the item or nothing. Refund usually contains currency.
+    const winnings = makePromise();
+    const refund = makePromise();
+    // Deposit contains buyer's payment, which is returned via refund if we did
+    // not win, or is split between earnings and refund if we did get the goods.
+    const deposit = makePromise();
+
+    // This promise will hold the seller's proceeds. It is ignored when this is
+    // not the winner's agency. When this seat corresponds to the auction's
+    // winner the seller receives funds here.
+    const earnings = makePromise();
+
+    const agencySeat = harden({
+      // The agency can accept one offer and collect the buyer's price or less.
+      // The buyer will receive their winnings through a trusted escrow.
+      consummateDeal(originalOffer, finalPrice, goodsPayment) {
+        const wonGoodsPayment = E(goodsIssuer).claimAll(goodsPayment, 'wins');
+        const { issuer: currencyIssuer } = currencyAmount.label;
+        const overbidAmount = E(currencyIssuer).makeAmount(
+          originalOffer - finalPrice,
```

Comment:
I think we shouldn't hardcode a Nat Strategy here. We could use `assay.without` or get the strategy from the issuer.